internal/kubernetes/pod_introspector_test.go

```diff
@@ -0,0 +1,437 @@
+package kubernetes_test
+
+import (
+	"context"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/internal/kubernetes"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+var _ = Describe("PodIntrospector", func() {
+	var (
+		ctx             context.Context
+		fakeClient      client.Client
+		testPodName     string
+		testNamespace   string
+		originalPodName string
+		originalNS      string
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		testPodName = "test-operator-pod"
+		testNamespace = "test-namespace"
+
+		// Store original environment variables
+		originalPodName = os.Getenv("POD_NAME")
+		originalNS = os.Getenv("NAMESPACE")
+
+		// Set up fake client with scheme
+		scheme := runtime.NewScheme()
+		err := corev1.AddToScheme(scheme)
+		Expect(err).ToNot(HaveOccurred())
+
+		fakeClient = fake.NewClientBuilder().WithScheme(scheme).Build()
+	})
+
+	AfterEach(func() {
+		// Restore original environment variables
+		if originalPodName != "" {
+			os.Setenv("POD_NAME", originalPodName)
```

Comment:
Not for this PR, but we should really create some kind of abstraction around this for testing at some point.